### PR TITLE
Update docstrings and clean up

### PR DESCRIPTION
**Pull request by shravan97**
In this PR : 
* Clean up docstrings and comments.
* Remove unnecessary `get_reducer_callable` from `CallableGenerator`.
* Make edits to `README.md`

### CODE REVIEW

File: PyRDF/__init__.py [hunk 1]
```diff
@@ -6,7 +6,21 @@
 includes = []
 
 def use(backend_name, conf = {}):
+    """
+    Chooses the backend of your choice
```
Comment (etejedor) [inline PyRDF/__init__.py]: Allows the user to choose the execution backend.

File: PyRDF/RDataFrame.py [hunk 2]
```diff
@@ -16,12 +16,15 @@ class RDataFrame(Node):
 
     Supported constructor arguments
     -------------------------------
-    - number of entries (int)
-        * For eg., RDataFrame(64)
-    - treename (str), fileslist (str or list of str)
-        * For eg., RDataFrame("myTree", "file.root") or RDataFrame("myTree", ["file1.root", "file2.root"])
-    - treename (str), fileslist (str or list of str or vector of str), default branches (list of str or vector of str)
-        * For eg., RDataFrame("myTree", ["file1.root", "file2.root"], ["branch1", "branch2"])
+    PyRDF's RDataFrame constructor is a replica of
```
Comment (etejedor) [inline PyRDF/RDataFrame.py]: PyRDF's RDataFrame constructor accepts the same arguments as the ROOT's RDataFrame constructors (see https://root.cern/doc/master/classROOT_1_1RDataFrame.html).
In addition, PyRDF allows to use Python lists in place of C++ vectors as arguments of the constructor, e.g. `RDataFrame("myTree", ["file1.root", "file2.root"]`.

File: PyRDF/__init__.py [hunk 3]
```diff
@@ -6,7 +6,21 @@
...
+    for execution of computational graph.
+
+    Parameters
+    ----------
+    backend_name : str
+        This is the name of the chosen backend.
 
+    conf (optional) : str
+        This should be a dictionary with necessary
+        configuration parameters. It's default value
```
Comment (etejedor) [inline PyRDF/__init__.py]: Its default value

File: PyRDF/backend/Backend.py [hunk 4]
```diff
@@ -57,17 +57,18 @@ def __init__(self, config={}):
...
 
-        Returns
-        -------
-        Operation.Types
-            The type of the current operation.
+        Raises
+        ------
+        Exception
+            This happens when operation_name doesn't
+            exist in supported_operations instance member.
```
Comment (etejedor) [inline PyRDF/backend/Backend.py]: `supported_operations` (with backticks)

File: PyRDF/Node.py [hunk 5]
```diff
@@ -86,6 +87,12 @@ def __setstate__(self, state):
         Retrieves the state dictionary of the current
         node and sets the instance variables.
 
+        Parameters
+        ----------
+        state : Python Dictionary
```
Comment (etejedor) [inline PyRDF/Node.py]: just dictionary?  Since we are in Python...

Comment (etejedor) [reply PyRDF/Node.py]: although if you change it, all the other docs should be consistent (also if it is a list)

**Comment by shravan97** [comment]
@etejedor , made the changes and rebased as well 👍

File: PyRDF/RDataFrame.py [hunk 6]
```diff
@@ -16,12 +16,10 @@ class RDataFrame(Node):
...
-    - number of entries (int)
-        * For eg., RDataFrame(64)
-    - treename (str), fileslist (str or list of str)
-        * For eg., RDataFrame("myTree", "file.root") or RDataFrame("myTree", ["file1.root", "file2.root"])
-    - treename (str), fileslist (str or list of str or vector of str), default branches (list of str or vector of str)
-        * For eg., RDataFrame("myTree", ["file1.root", "file2.root"], ["branch1", "branch2"])
+    PyRDF's RDataFrame constructor accepts the same arguments as the ROOT's RDataFrame
+    constructor (see https://root.cern/doc/master/classROOT_1_1RDataFrame.html). In addition,
+    PyRDF allows you to use Python lists in place of C++ vectors as arguments of the constructor,
+    for e.g., RDataFrame("myTree", ["file1.root", "file2.root"].
```
Comment (etejedor) [inline PyRDF/RDataFrame.py]: for e.g. is redundant, either for example or e.g.
Also, can we put the RDataFrame code in between backticks?

Comment (shravan97) [reply PyRDF/RDataFrame.py]: Sure @etejedor . Will change it right away 👍

**Comment by shravan97** [comment]
Done @etejedor 👍

**Pull request merged**